### PR TITLE
Added Snackbar qtquick object

### PR DESCRIPTION
**Pull request by ahcorde**
# 🎉 New feature

## Summary

Create a IgnSnackBar QtQuick object to show message to the user in the same manner by plugins

It could be :
 - persistant and then you need to click inside the GUI
 - or with a timeout

snackback

Added snackbar dialog to:
 - Save configuration
 - Teleop topic
 - Snapshot
 - Image view update

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: include/ignition/gui/MainWindow.hh [hunk 1]
```diff
@@ -568,6 +568,9 @@ namespace ignition
       /// \brief Displays a message to the user
       signals: void notify(const QString &_message);
 
+      /// \brief Displays a message to the user
+      signals: void notifyWithDuration(const QString &_message, int _duration);
```
Comment (chapulina) [inline include/ignition/gui/MainWindow.hh:572]: Missing `param` docs

File: include/ignition/gui/qml/IgnSnackBar.qml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  modal: duration == 0
+  focus: duration == 0
+  x: (window.width - width) / 2
+  y: window.height - window.height / 6
+  width: window.width - window.width / 6
+  contentHeight: notificationColumn.height
+
+  background: Rectangle {
+    border.color: "#444"
+    color: "#c9daf8"
```
Comment (chapulina) [inline include/ignition/gui/qml/IgnSnackBar.qml:35]: We should avoid using custom colors as much as possible and instead use colors from the current palette (`Material.primary`, `Material.accent`, `Material.background`). 

The [official spec](https://material.io/components/snackbars#anatomy) recommends a grey color, but I think QtQuick doesn't make it straight-forward to have that follow a dark / light theme. So I think we could stick to `Material.background` here.

File: src/plugins/teleop/Teleop.cc [hunk 3]
```diff
@@ -140,8 +140,19 @@ void Teleop::OnTopicSelection(const QString &_topic)
...
+    App()->findChild<MainWindow *>()->notifyWithDuration(
+      QString::fromStdString("Error when advertising topic: " +
+        this->dataPtr->topic), 1000);
+    ignerr << "Error when advertising topic: " <<
+      this->dataPtr->topic << std::endl;
+  }
+  else
+  {
+    App()->findChild<MainWindow *>()->notifyWithDuration(
+      QString::fromStdString("A new topic has been entered: '" +
```
Comment (chapulina) [inline src/plugins/teleop/Teleop.cc:153]: How about:

```suggestion
      QString::fromStdString("Subscribing to topic: '" +
```

File: include/ignition/gui/qml/IgnSnackBar.qml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  }
+
+  // Duration of the snackbar. If duration is equal to zero then
+  // you should click somewhere in Ignition Gazebo to close it.
+  property int duration: 1000
```
Comment (chapulina) [inline include/ignition/gui/qml/IgnSnackBar.qml:40]: I'm finding 1 second too short to read some of the snackbars on this PR. The Material spec [recommends](https://material.io/components/snackbars#behavior) "_They automatically disappear from the screen after a minimum of four seconds, and a maximum of ten seconds._", so how about defaulting to `4000` and using that throughout this PR?

**Review by chapulina — CHANGES_REQUESTED** [review]
+1 to the centralized approach using a signal from the `MainWindow` :+1: This should make it easier to improve the snackbar in the future. For example, in the future, if we start getting too many notifications at once, we may want to implement some logic to display them consecutively.

**Review by chapulina — APPROVED** [review]
LGTM! I just pushed a small change in eb58ef3aa3f05a6a379bfc2cb3052e709e658351 to change the border to a shadow.

Before:

border

After:

shadow

**Comment by osrf-triage** [comment]
This pull request has been mentioned on **Gazebo Community**. There might be relevant details there:

https://community.gazebosim.org/t/new-ignition-releases-2022-04-13-fortress-edifice/1367/1

**Pull request merged**